packages/augur-core/source/contracts/trading/ZeroXTrade.sol

```diff
@@ -220,15 +225,14 @@ contract ZeroXTrade is Initializable, IZeroXTrade, IERC1155 {
                 continue;
             }
 
-            uint256 _amountTraded = doTrade(_order, totalFillResults.takerAssetFilledAmount, _affiliateAddress, _tradeGroupId, msg.sender);
-
+            uint256 _amountTraded  = doTrade(_order, totalFillResults.takerAssetFilledAmount, _affiliateAddress, _tradeGroupId, _taker, _extraData);
             _fillAmountRemaining = _fillAmountRemaining.sub(_amountTraded);
         }
 
         transferFromAllowed = false;
 
         if (address(this).balance > 0) {
-            msg.sender.transfer(address(this).balance);
+            address(uint160(_taker)).transfer(address(this).balance);
```

Comment:
why uint160 and not just casting to address?

Reply:
`address(uint160())` makes it `address payable` which is required for `.transfer`.